assets/chezmoi.io/docs/reference/command-line-flags/common.md

```diff
@@ -6,10 +6,21 @@ The following flags apply to multiple commands where they are relevant.
 
 Set the output format.
 
+## `-h`, `--help`
```

Comment:
`--help` should not be moved from `global.md` as it *always* exists for all commands. Common flags apply to multiple different commands, but not to *all* commands.

Reply:
It can be added to both, I guess. Here is the output for `chezmoi apply --help` for example: https://gist.github.com/KapJI/049b54326d0aec676b3ea49c573fa92a

You can see that `--help` is subcommand flag, not global.

Reply:
The list of global and command flags from `--help` is programmatically determined by Cobra but that output should not necessarily be the determining factor on where something belongs which is a human judgement. Of the items in `chezmoi --help` listed as flags (which would be truly global), there are a number of things which are global because they make reasoning about *configuration* within chezmoi easier (such as `--interactive`) but are meaningless or useless for a large number of commands.

Picking on `--interactive`, this would include `age`, `archive`, `cat`, `cat-config`, `cd`, `completion`, `data`, `decrypt`, `destroy` (interactive by default), `diff`, `doctor`, `dump`, `dump-config`, `edit`, `edit-config`, `edit-config-template`, `encrypt`, `execute-template`, `forget` (interactive by default), `generate`, `git`, `help`, `ignored`, `license`, `managed`, `merge`, `merge-all`, `secret`, `source-path`, `state`, `status`, `target-path`, `unmanaged`, `upgrade`, and `verify`. That's all but five or six of the commands.

Documentation is about recording what matters, not merely what is.

Reply:
I see your point. In this case such flags should be moved to `common.md` but also added to subcommands where they matter. But maintaining such list looks pretty difficult and requires reading implementation of each particular command, so I don't think this is feasible in the long run.

This also creates different definitions for global flags. One according to implementation, another one according to documentation. This is confusing.

The ideal solution would be moving such arguments from global flags to subcommand flags. This will be benefitial for `--help` output and for documentation. But this is breaking change because some commands will stop accepting arguments they used to accept before, which is reasonable but also not great.

In the meantime, I think these global flags according to documentation should remain in `global.md`. Probably it's worth adding a note that not every flag has effect for every subcommand.